packages/cli/README.md

```diff
@@ -81,55 +81,169 @@ The above code shows you the use of the API and various objects and is a great w
+info
+info.initMsg
 
 // see your balance here
-smartQuery(client, addr, { balance: { address: faucetAddress } })
-
-// make a new contract
-const initMsg = { name: "Foo Coin", symbol: "FOO", decimals: 2, initial_balances: [{address: faucetAddress, amount: "123456789"}]}
-const foo = await instantiateContract(client, pen, 1, initMsg);
+smartQuery(client, addr, { balance: { address } })
+```
 
-smartQuery(client, foo, { balance: { address: faucetAddress } })
+Instantiate and use ERC20 contract:
 
-const rcpt = await randomAddress();
+```ts
+// no money? no problem.
+// let's make our own s**coin - replace "FOO" with something else to avoid duplicates
+const initMsg = { name: "Foo Coin", symbol: "FOO", decimals: 2, initial_balances: [{address, amount: "123456789"}]}
+const foo = await client.instantiate(1, initMsg, "FOO");
+foo
+foo.logs[0].events[0]
+const fooAddr = foo.contractAddress;
+
+// we can also find this another way...
+const fooAddr2 = await client.getContracts(1).then(contracts => contracts.filter(x => x.label == "FOO").map(x => x.address)[0])
+[fooAddr, fooAddr2]
+
+// now we have some cash
+smartQuery(client, fooAddr, { balance: { address } })
+
+const rcpt = await randomAddress("cosmos");
 rcpt
-smartQuery(client, foo, { balance: { address: rcpt } })
+smartQuery(client, fooAddr, { balance: { address: rcpt } })
 
 const execMsg = { transfer: {recipient: rcpt, amount: "808"}}
-const exec = await executeContract(client, pen, foo, execMsg);
+const exec = await client.execute(fooAddr, execMsg);
 exec
-exec[0].events[0]
-smartQuery(client, foo, { balance: { address: rcpt } })
+exec.logs[0].events[0]
+smartQuery(client, fooAddr, { balance: { address: rcpt } })
+```
+
+Or just send tokens:
+
+```ts
+client.getAccount(rcpt)
+
+const sent = await client.sendTokens(rcpt, [{amount: "1234", denom: "ucosm"}])
+sent
+foo.logs[0].events[0]
 ```
 
-## Other example codes
+### Use Custom Network
+
+All the network info can be configured inside the last argument to connect.
+To see how to connect to the Regen Testnet, try this. (Note you need to use `.editor`
+in the repl to allow multi-line commands. Alternative is to place entire `regenOptions`
+on one line.
 
-### Create random mnemonic and Cosmos address
+Run `./bin/cosmwasm-cli --init examples/helpers.ts`
 
 ```ts
-const mnemonic = Bip39.encode(Random.getBytes(16)).toString();
-const pen = await Secp256k1Pen.fromMnemonic(mnemonic);
-const pubkey = encodeSecp256k1Pubkey(pen.pubkey);
-const address = pubkeyToAddress(pubkey, "cosmos");
+.editor
+const regenOptions = {
+  httpUrl: "https://regen-lcd.vitwit.com/",
+  networkId: "kontraua",
+  feeToken: "utree",
+  gasPrice: 0.025,
+  bech32prefix: "xrn:",
+}
+^D
+
+const mnemonic = loadOrCreateMnemonic("regen.key");
+const {address, client} = await connect(mnemonic, regenOptions);
+address
+
+// check some random genesis account
+client.getAccount("xrn:1pdfr7xuckj6lhdphdde6peres9ufwgpsv87mag")
```

Comment:
`await` missing?

Reply:
Empty promises get resolved by the cli. 
Only needed when assigning

Reply:
Interesting, thanks